DLEDotNet/Editor/EditorStateBinder.cs

```diff
@@ -507,15 +535,32 @@ public void BindComboBox(ComboBox comboBox, string listProperty, string selected
         /// <param name="comboBox">The combo box to bind.</param>
         /// <param name="property">The name of the property under EditorState to bind; dots are allowed for nested properties. The property must have an enum type T.</param>
         public void BindComboBox<T>(ComboBox comboBox, string property) where T : Enum
+        {
+            BindComboBox<T>(comboBox, property, null);
+        }
+
+        /// <summary>
+        /// Binds a combo box to a (public) property with an enum type and a display text provider.
+        /// </summary>
+        /// <typeparam name="T">The type of the enum.</typeparam>
+        /// <param name="comboBox">The combo box to bind.</param>
+        /// <param name="property">The name of the property under EditorState to bind; dots are allowed for nested properties. The property must have an enum type T.</param>
+        /// <param name="displayTextProvider">A function that can provide a display text for an enum value.</param>
+        public void BindComboBox<T>(ComboBox comboBox, string property, GetEnumDisplayText<T> displayTextProvider) where T : Enum
         {
             comboBox.Items.Clear();
-            comboBox.DataSource = Enum.GetValues(typeof(T));
+            bool sorted = comboBox.Sorted;
+            comboBox.Sorted = false;
+            comboBox.DataSource = GetEnumAsList(typeof(T), displayTextProvider, sorted);
+            comboBox.DisplayMember = "Key";
+            comboBox.ValueMember = "Value";
+            comboBox.SelectedValue = LibDescent.Data.SegFunction.RedGoal;

```

Comment:
This looks like temporary debugging code that got left in - I'm guessing it's probably safe to delete line 557 since the property changed handler should be executed immediately anyway?